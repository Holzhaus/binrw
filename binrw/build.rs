use std::env;

fn main() {
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_SEARCH");
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_LIB");
    println!("cargo:rerun-if-changed=build.rs");

    if cfg!(bintrace) {
        if let Ok(link_search) = env::var("BINTRACE_LINK_SEARCH") {
            println!("cargo:rustc-link-search={}", link_search)
        }

        if let Ok(link_lib) = env::var("BINTRACE_LINK_LIB") {
            println!("cargo:rustc-link-lib={}", link_lib);
        } else {
            println!("cargo:warning=No bintrace backend provided");
        }
    }
}
