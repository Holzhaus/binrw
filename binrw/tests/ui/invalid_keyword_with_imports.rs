#![feature(generic_associated_types)]
use binrw::BinRead;

#[derive(BinRead)]
#[br(import { in_var: i16 })]
#[br(invalid_struct_keyword)]
struct Test {
    a: i16,
}

fn main() {
    // There should be no error from the compiler that the passed argument type
    // is wrong, since parsing of the struct was successful enough that the type
    // information could be resolved
    Test::read_args(
        &mut binrw::io::Cursor::new(b"\0\0"),
        <Test as BinRead>::Args::<'_>::builder()
            .in_var(1)
            .finalize()
    ).unwrap();
}
