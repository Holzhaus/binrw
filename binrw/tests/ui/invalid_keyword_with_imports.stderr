error: expected one of: `big`, `little`, `is_big`, `is_little`, `map`, `try_map`, `magic`, `import`, `import_raw`, `assert`, `pre_assert`
 --> $DIR/invalid_keyword_with_imports.rs:5:6
  |
5 | #[br(invalid_struct_keyword)]
  |      ^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `builder` found for trait object `(dyn for<'arg> ArgType<'arg, for<'arg> Item = TestBinReadArgs> + 'static)` in the current scope
  --> $DIR/invalid_keyword_with_imports.rs:16:34
   |
16 |         <Test as BinRead>::Args::builder()
   |                                  ^^^^^^^ function or associated item not found in `(dyn for<'arg> ArgType<'arg, for<'arg> Item = TestBinReadArgs> + 'static)`
